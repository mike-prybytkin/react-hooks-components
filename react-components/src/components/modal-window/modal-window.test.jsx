import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import ModalWindow from './modal-window';

const handleClose = jest.fn();
const setUp = () =>
  render(
    <ModalWindow onModalWindow={handleClose}>
      <div>test</div>
    </ModalWindow>
  );

describe('ModalWindow component', () => {
  it('should render modal window with children', () => {
    setUp();
    expect(screen.getByText('test')).toBeInTheDocument();
  });

  it('should close modal window', () => {
    setUp();
    fireEvent.click(screen.getByText(/âœ—/i));
    expect(handleClose).toHaveBeenCalledTimes(1);
  });

  it('create modal window snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
