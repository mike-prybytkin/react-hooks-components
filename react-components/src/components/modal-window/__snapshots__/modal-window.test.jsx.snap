// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWindow component create modal window snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal-overlay"
  >
    <div
      class="modal-window"
    >
      <button
        class="modal-window__close"
      >
        âœ—
      </button>
      <div
        class="modal-window__body"
      >
        <div>
          test
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
